site-permissions/site-permissions-impl/src/main/java/com/duckduckgo/site/permissions/impl/SitePermissionsDialogActivityLauncher.kt

```diff
@@ -108,6 +132,87 @@ class SitePermissionsDialogActivityLauncher @Inject constructor(
             .show()
     }
 
+    private fun showSiteDrmPermissionsDialog(
+        activity: Activity,
+        url: String,
+        tabId: String,
+        request: PermissionRequest,
+    ) {
+        val domain = url.extractDomain() ?: url
+
+        // Check if user allowed or denied per session
+        val sessionSetting = sitePermissionsRepository.getDrmForSession(domain)
```

Comment:
Originally we thought to keep this tied to a `tabId` as well, but since each tab has its own `SitePermissionsDialogActivityLauncher`, we can skip that. This does seem a bit hacky to me - perhaps we can design a better mechanism for session-based permissions once we revisit the 24hr based-ones (camera/mic).

Reply:
It seems easy for someone to use `sitePermissionsRepository.getDrmForSession` outside of the corresponding `SitePermissionsDialogActivityLauncher` and start getting inconsistent results.

Let's wait for the Privacy Triage meeting tomorrow to see if we'll revert back to 24h permission.